### ISSUE TITLE
Problem to install @kyleshockey/js-yaml dependence

### ISSUE THREAD

**Issue by ponessa** [swagger-api/swagger-ui#5362]
### Q&A (please complete the following information)
 - OS: macOS Mojava v10.14.4
 - Browser: Chrome and Firefox
 - Version: Chrome 74.0.3729.157 or Firefox 60.6.3ers
 - Method of installation: npm
 - Swagger-UI version: latest
 - Swagger/OpenAPI version: Swagger 2.0

### Describe the bug you're encountering
Hello,  I am trying to install the IBM APIConnect product which includes @kyleshockey/js-yaml@1.0.1.  I continue to receive the 404 error below.
```
npm ERR! code E404
npm ERR! 404 Not Found: @keleshockey/js-yaml@1.0.1
```
I have tried the resolution for [closed issue #4941](https://github.com/swagger-api/swagger-ui/issues/4941){:target="_blank"}, however, still receive the 404 error whether trying to install `apiconnect` or `@kyleshockey/js-yaml@1.0.1` (e.g. `npm install -g @kyleshockey/js-yaml@1.0.1`).

Note that apiconnect requires node v8.15.1 or higher but below v9.*.  Installing the latest npm puts me at v6.9.0.

### To reproduce...

```
npm install -g @kyleshockey/js-yaml@1.0.1
```
Output
```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@kyleshockey%2fjs-yaml - Not Found
npm ERR! 404 
npm ERR! 404  '@kyleshockey/js-yaml@1.0.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/steve/.npm/_logs/2019-05-19T18_05_03_430Z-debug.log
```

**Comment by shockey** [comment]
@ponessa what npm version are you using? are you using a private registry? are you behind a firewall?

**Comment by shockey** [comment]
FWIW, this works on my system, with npm 6:

```
➜ npm -v
6.4.1
➜ npm install -g @kyleshockey/js-yaml@1.0.1
/Users/kyle/.nvm/versions/node/v10.15.3/bin/js-yaml -> /Users/kyle/.nvm/versions/node/v10.15.3/lib/node_modules/@kyleshockey/js-yaml/bin/js-yaml.js
+ @kyleshockey/js-yaml@1.0.1
added 3 packages from 8 contributors in 1.691s
```

**Comment by ponessa** [comment]
Thanks.  I fully understand .. seems to work for a bunch of people but me.
```
Stevens-MacBook-Pro:proof-o-concept steve$ npm -v
6.9.0
Stevens-MacBook-Pro:proof-o-concept steve$ npm install -g @kyleshockey/js-yaml@1.0.1
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@kyleshockey%2fjs-yaml - Not Found
npm ERR! 404 
npm ERR! 404  '@kyleshockey/js-yaml@1.0.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/steve/.npm/_logs/2019-05-21T02_35_25_271Z-debug.log
```
I have also tried with my VPN both off and on to see if it was something my company was blocking.  I believe it has something to do with scoped packages, not the @kyleshockey@js-yaml package specifically.

**Comment by shockey** [comment]
@ponessa yeah, there's definitely something funky on your end... the URL that your error message mentions actually _does_ exist:
```
➜ curl -s -o /dev/null -v https://registry.npmjs.org/@kyleshockey%2fjs-yaml
> GET /@kyleshockey%2fjs-yaml HTTP/2
> Host: registry.npmjs.org
> User-Agent: curl/7.54.0
> Accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 200 
< date: Tue, 21 May 2019 21:56:12 GMT
< content-type: application/json; charset=UTF-8
< set-cookie: __cfduid=de94e335f7561b76fd96620d51129fd631558475772; expires=Wed, 20-May-20 21:56:12 GMT; path=/; domain=.registry.npmjs.org; HttpOnly
< cf-ray: 4da9cc897e0f991d-LAX
< age: 166
< etag: W/"5b3d6ac7-398d"
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< last-modified: Thu, 5 Jul 2018 0:48:07 GMT
< vary: Accept, Accept-Encoding
< via: 1.1 varnish (Varnish/5.2)
< x-varnish: 238314432 240938347
< server: cloudflare
< 
{ [986 bytes data] }
```

**Comment by smartmouse** [comment]
I cannot reproduce in my mac as well which is  
<img width="562" alt="Screen Shot 2019-05-27 at 1 56 44 PM" src="https://user-images.githubusercontent.com/4519627/58433949-4a7a9a80-8087-11e9-8fdd-d7830a3cb4f4.png">

could this be specific to the mac OS? or your specific environment?
Can you delete your cache and try again?
Are you using nvm?

**Comment by smartmouse** [comment]
if you do a npm info on the scope package, can u even see it in your env?

**Comment by theodu60** [comment]
Hi,
Do you have any .npmrc file ?
If you do have this file in your home folder, try to remove it.
( i use a Mac with Mojave, and this file was in: /Users/MYUSER/.npmrc )

**Comment by ponessa** [comment]
Hi,

We were finally able to resolve the issue by NOT using NVM.

To get a clean environment, I tried the installation on a docker image.  Although I received other errors, the scoped package installed.  So I got rid Need to get rid of nvm by commenting out NVM entries from .bash_profile and renaming $HOME/.npmrc to .npmrc.bak and .npm to .npm.bk.

Thereafter:
npm install -g apiconnect --no-optional --production

Worked.

I can't explain it but it does look like the problem was with NVM.  Thank you to all that commented.  I am going to leave this open for a few days to see if someone might understand why NVM was causing this problem.  And thanks again for all the suggestions, they were helpful in finally figuring out what the problem was.
